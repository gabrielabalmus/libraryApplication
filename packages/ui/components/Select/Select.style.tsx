import Select from "@mui/material/Select";
import { styled as MuiStyled } from "@mui/material/styles";
import { SelectFieldProps } from "./Select.type";

export const StandardSelect = MuiStyled(Select)<SelectFieldProps>(
  ({ width, theme }) =>
    theme.unstable_sx({
      width: "100%",
      maxWidth: { md: width, xs: "100%" },
    })
);
