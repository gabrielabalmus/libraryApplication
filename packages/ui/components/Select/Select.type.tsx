import { SelectProps } from "@mui/material/Select";

export interface SelectFieldProps extends SelectProps<string> {
  width?: string;
}

export interface SelectContainerProps {
  label?: string;
  onChange: (value: string) => void;
  value: string;
  options: SelectOptions[];
  errorMessage?: string;
  width?: string;
  placeholder?: string;
  disabled?: boolean;
}

export interface SelectOptions {
  value: string;
  name: string;
}
