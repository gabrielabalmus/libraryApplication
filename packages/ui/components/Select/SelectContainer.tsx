import FormControl from "@mui/material/FormControl";
import { StandardSelect } from "./Select.style";
import { SelectContainerProps } from "./Select.type";
import MenuItem from "@mui/material/MenuItem";
import InputLabel from "@mui/material/InputLabel";
import FormHelperText from "@mui/material/FormHelperText";

const SelectContainer: React.FC<SelectContainerProps> = ({
  label,
  value,
  options,
  errorMessage,
  onChange,
  width = "inherit",
  placeholder = "",
}) => {
  const handleOnChange = (event: any) => {
    onChange(event.target.value);
  };

  return (
    <FormControl variant="standard" error={!!errorMessage}>
      {label && <InputLabel>{label}</InputLabel>}
      <StandardSelect
        value={value}
        width={width}
        onChange={handleOnChange}
        variant="standard"
        placeholder={placeholder}
      >
        {options.map((item, index) => (
          <MenuItem key={index} value={item.value}>
            {item.name}
          </MenuItem>
        ))}
      </StandardSelect>
      {errorMessage && <FormHelperText>{errorMessage}</FormHelperText>}
    </FormControl>
  );
};

export default SelectContainer;
