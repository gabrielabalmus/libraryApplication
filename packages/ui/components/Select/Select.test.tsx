import { fireEvent, render, screen } from "@testing-library/react";
import Select from ".";
import { SelectContainerProps, SelectOptions } from "./Select.type";
import { useState } from "react";
import "@testing-library/jest-dom";

const mockChange = jest.fn();

const options: SelectOptions[] = [
  {
    value: "option1",
    name: "Option1",
  },
  {
    value: "option2",
    name: "Option2",
  },
  {
    value: "option3",
    name: "Option3",
  },
];
const renderSelect = ({ ...props }: SelectContainerProps) =>
  render(<Select {...props} />);

describe("snapshot tests for Select", () => {
  it("renders correctly", () => {
    const { container } = renderSelect({
      label: "Test1",
      value: "option1",
      onChange: mockChange,
      width: "120px",
      options,
    });

    expect(container).toMatchSnapshot();
  });

  it("renders correctly with error message", () => {
    const { container } = renderSelect({
      label: "Test1",
      value: "option1",
      onChange: mockChange,
      options,
      errorMessage: "Error message",
    });

    expect(container).toMatchSnapshot();
  });

  it("renders correctly with placeholder", () => {
    const { container } = renderSelect({
      label: "Test1",
      value: "option1",
      onChange: mockChange,
      options,
      placeholder: "Select placeholder",
    });

    expect(container).toMatchSnapshot();
  });
});

const SelectState = (
  other: Omit<SelectContainerProps, "onChange" | "options">
) => {
  const [value, setValue] = useState<string>(other.value);

  return (
    <Select {...other} value={value} onChange={setValue} options={options} />
  );
};

const wrapper = (other: Omit<SelectContainerProps, "onChange" | "options">) =>
  render(<SelectState {...other} />);

describe("logic tests for Select", () => {
  it("takes value correctly", () => {
    wrapper({ value: "option1", placeholder: "Test placeholder" });

    const select = screen.getByPlaceholderText("Test placeholder");
    expect(select).toBeInTheDocument();
    expect(select).toHaveValue("option1");

    const value = "option2";
    fireEvent.change(select, {
      target: {
        value,
      },
    });
    expect(select).toHaveValue(value);
  });
});
