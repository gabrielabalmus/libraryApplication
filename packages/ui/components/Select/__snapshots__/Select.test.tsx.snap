// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for Select renders correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Test1
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-h9cx5f-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      width="120px"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        role="button"
        tabindex="0"
      >
        Option1
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        placeholder=""
        tabindex="-1"
        value="option1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`snapshot tests for Select renders correctly with error message 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Test1
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-1sptybj-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      width="inherit"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        role="button"
        tabindex="0"
      >
        Option1
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        placeholder=""
        tabindex="-1"
        value="option1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
    >
      Error message
    </p>
  </div>
</div>
`;

exports[`snapshot tests for Select renders correctly with placeholder 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Test1
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1sptybj-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      width="inherit"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        role="button"
        tabindex="0"
      >
        Option1
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        placeholder="Select placeholder"
        tabindex="-1"
        value="option1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
